addons/account/models/account_move.py

```diff
@@ -4318,7 +4318,7 @@ def _prepare_analytic_distribution_line(self, distribution):
             'ref': self.ref,
             'move_id': self.id,
             'user_id': self.move_id.invoice_user_id.id or self._uid,
-            'company_id': distribution.account_id.company_id.id or self.env.company.id,
+            'company_id': distribution.account_id.company_id.id or self.move_id.company_id.id or self.env.company.id,
```

Comment:
```suggestion
            'company_id': distribution.account_id.company_id.id or self.company_id.id or self.env.company.id,
```
This doesn't change much, but you the company of the move can directly be found on the line too